WtWSMS/common/on_action/dlc/ep1/BP_court_language_on_actions.txt

```diff
@@ -46,3 +43,214 @@ bp_court_language_on_actions = {
+			}
+			set_court_language = language_sanskrit
+		}
+	}
+}
+
+on_game_start = {
+	on_actions = { setup_administrative_languages }
+}
+
+setup_administrative_languages = {
+	effect = {
+		#Set some court langauge variables for bureaucrats
+		title:e_wre = {
+			set_variable = {
+				name = court_language
+				value = culture:roman
+			}
+		}
+		title:e_byzantium = {
+			if = {
+				limit = { current_date < 610.10.5 }
+				set_variable = {
+					name = court_language
+					value = culture:roman
+				}
+			}
+			else = {
+				set_variable = {
+					name = court_language
+					value = culture:greek
+				}
+			}
+		}
+		title:k_italy = {
+			set_variable = {
+				name = court_language
+				value = culture:roman
+			}
+			holder = {
+				setup_the_court_language_effect = {
+					CHAR = this
+					LANGUAGE = language_latin
+				}
+			}
+		}
+		#Byzantines use Latin before Hearclius' reign
+		#Source: Davis, Leo Donald (1990). The First Seven Ecumenical Councils (325–787): Their History and Theology. Liturgical Press. p. 260. 
+		every_ruler = {
+			limit = { has_title = title:e_byzantium }
+			if = {
+					limit = { current_date < 610.10.5 }
+					setup_the_court_language_effect = {
+						CHAR = this
+						LANGUAGE = language_latin
+					}
+				}
+				else = { #Otherwise, learn Greek
+					if = {
+						limit = { primary_title = { tier >= tier_kingdom } }
+						setup_the_court_language_effect = {
+							CHAR = this
+							LANGUAGE = language_greek
+						}
+					}
+					else = {
+						setup_learnt_language_effect = {
+							CHAR = this
+							LANGUAGE = language_greek
+						}
+					}
+				}
+			every_vassal_or_below = {
+				limit = { has_government = gubernatorial_government }
+				if = {
+					limit = { current_date < 610.10.5 }
+					if = {
+						limit = { primary_title = { tier >= tier_kingdom } }
+							setup_the_court_language_effect = {
+							CHAR = this
+							LANGUAGE = language_latin
+						}
+					}
+					else = {
+						setup_learnt_language_effect = {
+							CHAR = this
+							LANGUAGE = language_latin
+						}
+					}
+				}
+				else = { #Otherwise, learn Greek
+					if = {
+						limit = { primary_title = { tier >= tier_kingdom } }
+						setup_the_court_language_effect = {
+							CHAR = this
+							LANGUAGE = language_greek
+						}
+					}
+					else = {
+						setup_learnt_language_effect = {
+							CHAR = this
+							LANGUAGE = language_greek
+						}
+					}
+				}
+			}
+		}
+		#Gupta Governments use Sanskrit
+		every_ruler = {
+			limit = { has_government = rajamandala_government }
+			if = {
+				limit = { primary_title = { tier >= tier_kingdom } }
+				setup_the_court_language_effect = {
+					CHAR = this
+					LANGUAGE = language_sanskrit
+				}
+			}
+			else = { #Ensure every Gupta ruler knows Sanskrit
+				setup_learnt_language_effect = {
+					CHAR = this
+					LANGUAGE = language_sanskrit
+				}
+			}
+		}
+		#Sub-Romans (in 476 at least) use Latin
```

Comment:
Are we sure on this one, especially in the Eastern half of the empire? I think it should probably take into account Greek. And in areas like Britain rulers using Latin after Roman withdrawal is uncertain

Reply:
I mean, Greek was adopted in 620, which (in CK3 parlance) would make it the court language, no?

By default, newly minted Kingdoms use the language of the ruler as the court language. Court languages are only used by Kingdoms and the only Sub-Roman polity in Britain is the duchy Ambrosius possesses (so no Royal Court and no Court Language).

![20231030113458_1](https://github.com/loup99/BP/assets/55886300/02d2908c-6c42-4a46-a412-1cef486504e1)

Reply:
My bad, I thought it applied to duchies as well since I don't have the DLC. Didn't realise it was only kingdoms although I guess "Royal" Court should have led me to guess it.